client/ayon_core/plugins/publish/integrate.py

```diff
@@ -792,7 +792,7 @@ def prepare_representation(
         # Explicitly store the full list even though template data might
         # have a different value because it uses just a single udim tile
         if repre.get("udim"):
-            repre_context["udim"] = repre.get("udim")  # store list
+            repre_context["udim"] = repre.get("udim")[0]  # store list
```

Comment:
The comment above this scares me so much - why did I do this? :)

Anyway, I believe that technically we should be able to just take the value from the `template_data` because that would've always been the `udim` individual tile that formatted the path. Which is what this value should've been to begin with.

Only thing somewhat related I can find is:
- this comment discussion: https://github.com/ynput/OpenPype/pull/2898#discussion_r829199778
- https://github.com/ynput/OpenPype/pull/2898/commits/b128e0addffc77991e5ff25f2d219d8ed8613136
- https://github.com/ynput/OpenPype/pull/2898#issuecomment-1092124350

Reply:
Having a look at the code again I think unfortunately, how odd it is - this may be the correct fix. I can't find any other usage of the `udim` key in which this seems to become problematic at first sight.

Only other thing that I thought of is that we could also just remove these lines completely because the `udim` key should already be in `repre_context["udim"]` due to them being used values in the template. The only thing keeping this line around with your changes does is make sure the `udim` is also in the context if it were not used by the template. Whether that'd be correct or not behavior @iLLiCiTiT will know - going based of the links in my above comment he had some opinions on that, but looking back now and re-reading that I'm still not sure I fully understood what he concluded it at.